### DISCUSSION TITLE
iPad/Tablet - contained layout for bigger screen devices

### DISCUSSION THREAD

**Discussion by thupi** [gorhom/react-native-bottom-sheet#234] [Ideas]
# Feature Request

  This is a great package, especially due to the great support React Navigation and high performence. However, one missing feature seems to be the abilitiy to configure the BottomSheet to behave native-like on tablets and big screen devices. 
  
With native-like i mean like the following:

![image](https://user-images.githubusercontent.com/16625262/105488624-3353dc00-5cb2-11eb-8e6c-849e38eac9ed.png)

## Why it is needed

We need this feature in order to deliver a native-like experience on big screen devices. This package seems to get a lot of thing right but would benefit from having the option to configure the sheet to be contained, more like the native implementations such as in the AppStore as shown above :-)

## Possible implementation

I imagine that we could add an option to define the maxWidth and height of the sheet. That way the sheet wouldn't go all the way to the edges but stay centered. The last part would be to have some way of telling the sheet to open to the centered state. Could be achived by either just keep using the % definition or support a new method that would open the sheet to the full size.

### Code sample

Not really sure which changes would be needed to achive something like this yet.

Looking forward to hear what you guys think of the idea :-)

**Comment by gorhom** [comment]
thanks @thupi for submitting this feature request ! this is indeed something we need to implement here.

currently i am busy with the keyboard handling #203 , but i would like to see others to get involve with this too

**Comment by thupi** [comment]
Hi @gorhom, 

Just had a bit of time to look into this. Seems to me like there are 2 separate parts to this feature.

1. Contain the width of the sheet
As of now, the sheet seems to always fill the entire space of the container. So i added added a sheetMaxWidth prop. 

It should require a fairly small amount of changes to the container style:
```
const containerContainedStyle = useMemo(
  () => ({
    maxWidth: _providedSheetMaxWidth,
    left:
      _providedSheetMaxWidth && _providedSheetMaxWidth < safeContainerWidth
        ? (safeContainerWidth - _providedSheetMaxWidth) / 2
        : 0,
  }),
  [_providedSheetMaxWidth, safeContainerWidth]
);
```

I was wondering if it would make sense to just use `WINDOW_WIDTH` for `safeContainerWidth` or if it should be measured the same way we measure the height. 

2. Contain the height of the sheet
Containing the height seems to be a bit more complicated as height is referenced many places and i need to get an overview how sheet height is related to other functionality.

Luckily these 2 things might not be codependent so i guess i could start with the 1. :-) 

Any input here :-) ?

**Reply by gorhom** [reply]
hi @thgh , sorry for replying late 🙏

> Contain the width of the sheet As of now, the sheet seems to always fill the entire space of the container. So i added added a sheetMaxWidth prop.

yeah we could introduce this prop to limit the sheet width but not the container, because the container also hold the backdrop.

> I was wondering if it would make sense to just use WINDOW_WIDTH for safeContainerWidth or if it should be measured the same way we measure the height.

I think we need to have `safeContainerWidth` , because we don't know if user going to place `BottomSheet` at the root of the screen or wrapped with view with horizontal padding.

> Containing the height seems to be a bit more complicated as height is referenced many places and i need to get an overview how sheet height is related to other functionality.

it is, the actual sheet height would be `sheetHeight` and i don't think that it needs to change.

you could give action 1 a go, and then we could discuss the `detach` sheet step.

**Comment by nandorojo** [comment]
Just checking what the status is on this feature. It would be a great addition.

**Comment by noskovvkirill** [comment]
Curious if there is any progress on this 👀

**Comment by Brma1048** [comment]
@gorhom any updates on this?
